posts/2024-10-17-Rust-1.82.0.md

```diff
@@ -0,0 +1,320 @@
+
+<pre>
+<b><span style="color:#0A0">cc</span></b> <b><span style="color:#0AA">#build-dependencies</span></b>
+A build-time dependency for Cargo build scripts to assist in invoking the native
+C compiler to compile native C code into a static archive to be linked into Rust
+code.
+<b><span style="color:#0A0">version:</span></b> 1.1.23 <b><span style="color:#A50">(latest 1.1.30)</span></b>
+<b><span style="color:#0A0">license:</span></b> MIT OR Apache-2.0
+<b><span style="color:#0A0">rust-version:</span></b> 1.63
+<b><span style="color:#0A0">documentation:</span></b> https://docs.rs/cc
+<b><span style="color:#0A0">homepage:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">repository:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">crates.io:</span></b> https://crates.io/crates/cc/1.1.23
+<b><span style="color:#0A0">features:</span></b>
+  jobserver = []
+  parallel  = [dep:libc, dep:jobserver]
+<b><span style="color:#0AA">note</span></b><b>:</b> to see how you depend on cc, run `<b><span style="color:#0AA">cargo tree --invert --package cc@1.1.23</span></b>`
+</pre>
+
+By default `cargo info` describes the package version in the local `Cargo.lock`, if any, but as you can see it will indicate when there's a newer version too, and `cargo info cc@1.1.30` would report on that. The feature list also indicates a `+` next to any that are enabled.
+
+### Apple target promotions
+
+#### macOS on ARM64 is now Tier 1
+
+The Rust target `aarch64-apple-darwin` for macOS on ARM64 (M1-family or later Apple Silicon CPUs) is now a tier 1 target, indicating our highest guarantee of working properly. As the [platform support](https://doc.rust-lang.org/stable/rustc/platform-support.html) page describes, every change in the Rust repository must pass full tests on every tier 1 target before it can be merged. This ARM64 target was added to tier 2 back in Rust 1.49, making it available in `rustup`, and this new milestone raises it on par with ARM64 Linux and the X86 macOS, Linux, and Windows targets.
+
+#### Mac Catalyst targets are now Tier 2
+
+[Mac Catalyst](https://developer.apple.com/mac-catalyst/) is a technology by Apple that allows running iOS applications natively on the Mac. This is especially useful when testing iOS-specific code, as `cargo test --target=aarch64-apple-ios-macabi --target=x86_64-apple-ios-macabi` mostly just works (in contrast to the usual iOS targets, which need to be bundled using external tooling before they can be run on a native device or in the simulator).
+
+[The targets](https://doc.rust-lang.org/nightly/rustc/platform-support/apple-ios-macabi.html) are now tier 2, and can be downloaded with `rustup target add aarch64-apple-ios-macabi x86_64-apple-ios-macabi`, so now is an excellent time to update your CI pipeline to test that your code also runs in iOS-like environments.
+
+### Minimal exhaustive patterns
+
+Empty patterns can now be omitted in common cases:
```

Comment:
It seems to me that this section heading isn't very clear. It's true that the feature name is `min_exhaustive_patterns`, but the name doesn't really communicate what this is, particularly because the obvious interpretation of “exhaustive” is in the sense that “`match`es must be exhaustive”, which is *already* true, so saying “[Rust now allows] exhaustive patterns” is odd.

```suggestion
### Omitting empty types in pattern matching

Patterns which match empty (a.k.a. uninhabited) types by value can now be omitted:
```

Reply:
I agree with all your comments on this section (I'm the one who wrote the original text) 👍